### ISSUE TITLE
Expose the QueryMetadata on the AbstractMongodbQuery so it can be converted to a Morphia Query

### ISSUE THREAD

**Issue by gaspardpetit** [querydsl/querydsl#1842]
I use QueryDSL not just for querying, but also for deleting and updating mongo entities.  Since QueryDSL only support queries, I have to convert the AbstractMongodbQuery into a Morphia query (or break it down into a DBObject and obtain query modifiers to talk directly using the java driver).

The only workaround I know to convert a QueryDSL to a Morphia Query / DBObject is a hack whereby the query is serialized to JSON, deserialized to a DBObject and then jacked into a Morphia query using reflection to call an otherwise private method:

See https://groups.google.com/forum/#!topic/querydsl/ItCdp5wR6Vs

This works in simple cases where there is no limit or sorting on the query, but we lose modifier information.

Would it be possible to expose a getMetadata() on AbstractMongodbQuery so that we can get the limit, order and projection modifiers?

Alternatively, a "toMorphiaQuery" method on the AbstractMongodbQuery would also be a good addition.

I wouldn't mind contributing with a patch - just let me know which solution seems more appropriate.

Gaspard

**Comment by timowest** [comment]
Mongodb delete doesn't support pagination and order so it might be enough to just get the query definition out?

**Comment by gaspardpetit** [comment]
You are making a very good point :)

The getAsDBObject is a great improvement.

There were still a few cases where I wanted to extract the full query (with options) - for example, to create a DBCursor on which I can then change the timeout (I did not find any way to change the timeout on QueryDSL iterator).  Logging the query details for debugging is also useful (the toString() doesn't give this information).  All in all, this is not a big deal - I realized I could get the queryMixingField using reflection, so for now that'll work. But if exposing the QueryMixin makes sense on your end, I can definitely see cases where I would pull information from it.

In any case, thank you for the change #1851 and your quick response!
